engine/runtime/src/test/java/org/enso/interpreter/data/test/HashMapTest.java

```diff
@@ -0,0 +1,196 @@
+package org.enso.interpreter.data.test;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+import com.oracle.truffle.api.interop.InteropLibrary;
+import java.io.ByteArrayOutputStream;
+import java.nio.file.Paths;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.concurrent.Callable;
+import org.enso.interpreter.node.expression.builtin.meta.EqualsAnyNode;
+import org.enso.interpreter.node.expression.builtin.meta.EqualsAnyNodeGen;
+import org.enso.interpreter.node.expression.builtin.meta.HashCodeAnyNode;
+import org.enso.interpreter.node.expression.builtin.meta.HashCodeAnyNodeGen;
+import org.enso.interpreter.runtime.data.hash.EnsoHashMap;
+import org.enso.interpreter.runtime.data.hash.EnsoHashMapBuilder;
+import org.enso.polyglot.RuntimeOptions;
+import org.graalvm.polyglot.Context;
+import org.graalvm.polyglot.Language;
+import org.graalvm.polyglot.Value;
+import org.graalvm.polyglot.proxy.ProxyExecutable;
+import org.junit.Before;
+import org.junit.Test;
+
+public class HashMapTest {
+  private InteropLibrary interop;
+  private HashCodeAnyNode hashCodeNode;
```

Comment:
If you need to verify behavior of `Any.hash_code` (that I'd like to see removed), then put such tests here and use the internals of the engine to access the hash via the `hashCodeNode`.